innopoints/core/notifications/push.py

```diff
@@ -0,0 +1,156 @@
+"""Helper module for sending notifications, depending on the user preference"""
+
+import logging
+
+from pywebpush import WebPushException
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm.attributes import flag_modified
+
+from innopoints.extensions import push as webpush, db
+from innopoints.models import NotificationType, Account, StockChangeStatus, ApplicationStatus
+from innopoints.schemas import PayloadSchema
+
+log = logging.getLogger(__name__)
+
+
+def get_content(notification_type, payload):
+    '''Given notification type and payload, returns the data for the notification'''
+    # default values
+    title = 'Innopoints notification'
+    body = 'Something happened.'
+    link = '/'
+    image = None
+    if notification_type == NotificationType.purchase_status_changed:
+        title = 'Purchase status changed'
+        if payload['product']['type'] is None:
```

Comment:
You sure this works? In the email module it was along the lines of `payload['product'].type`, because even though `payload` is a dictionary, it contains the actual model objects, which disallow dict-like property access

Reply:
Check line 129. I am working with dicts, not models